### PR TITLE
fix: added failing test data for issues weve been seeing with aliases

### PR DESCRIPTION
**Pull request by tomhobson**
Added a load of tests to demonstrate an issue I'd found.

Decided after to actually fix it too.

fixes: https://github.com/jenkins-x-plugins/jx-promote/issues/387

### CODE REVIEW

File: pkg/rules/helmfile/helmfile_rule.go [hunk 1]
```diff
@@ -36,7 +36,7 @@ func Rule(r *rules.PromoteRule) error {
 }
 
 // ModifyAppsFile modifies the 'jx-apps.yml' file to add/update/remove apps
-func modifyHelmfile(r *rules.PromoteRule, rule *v1alpha1.HelmfileRule, file, promoteNs string) error {
+func modifyHelmfile(r *rules.PromoteRule, rule *v1alpha1.HelmfileRule, file string, promoteNs string) error {
```
Comment (ankitm123) [inline pkg/rules/helmfile/helmfile_rule.go:39]: u dont need string here, so file, promoteNS string is the right thing, this will raise a linting issue (paramtypecombine from gocritic)

File: pkg/rules/helmfile/helmfile_rule.go [hunk 2]
```diff
@@ -209,8 +244,7 @@ func defaultPrefix(appsConfig *state.HelmState, envctx *envctx.EnvironmentContex
 	}
 	prefixes := map[string]string{}
 	urls := map[string]string{}
-	for k := range appsConfig.Repositories {
-		r := appsConfig.Repositories[k]
+	for _, r := range appsConfig.Repositories {
```
Comment (ankitm123) [inline pkg/rules/helmfile/helmfile_rule.go:247]: this change will also flag the linter, u can do r.URL, and that should work.

**Review by ankitm123 — COMMENTED** [review]
You have the test cases, it would be nice to add a test for this, as this is a critical piece, and regression here can be a problem.

I can spend some time writing tests, and we can work on this together, wdyt?

File: pkg/rules/helmfile/helmfile_rule.go (hunk 2 above)
Comment (tomhobson) [reply pkg/rules/helmfile/helmfile_rule.go]: Looks like goland was trying to be a bit too clever! Fixed

File: pkg/rules/helmfile/helmfile_rule.go (hunk 1 above)
Comment (tomhobson) [reply pkg/rules/helmfile/helmfile_rule.go]: Goland trying to be too clever, fixed!

**Comment by ankitm123** [comment]
/hold
putting on hold to write some tests.

File: pkg/rules/factory/factory_test.go [hunk 3]
```diff
@@ -30,7 +30,7 @@ func TestRuleFactory(t *testing.T) {
 
 	t.Logf("creating tests at %s", tmpDir)
 
-	sourceData := filepath.Join("test_data")
```
Comment (ankitm123) [inline pkg/rules/factory/factory_test.go:33]: Why this change?

**Review by ankitm123 — APPROVED** [review]

**Comment by ankitm123** [comment]
> /hold putting on hold to write some tests.

Not required, as the factory_test files test the helmfile.Rule function.

**Comment by tomhobson** [comment]
We're ready to go!
